python/lsst/pipe/base/graph/_implDetails.py

```diff
@@ -81,6 +85,11 @@ def addProducer(self, key: _T, value: _U):
         if self._createInverse:
             self._itemsDict[value].add(key)
 
+    def removeProducer(self, key: _T, value: _U):
```

Comment:
This is one of the things I added while you were working, I think it explains things to your satisfaction, but might make most sense in the context of understanding the container, which is generic, so its hard to be too specific.. I like the naming because it works well with existing add and remove/consumer.

Reply:
I would still like a clearer answer to the bulleted question; right now it reads like it fully removes (e.g.) the `QuantumNode` that produces a `DatasetRef`, which can't be right since it doesn't automatically remove that node's input-dataset links.

Reply:
I will need to think about how to reword this, because it does exactly what is says, it removes a value from being marked as a producer of the associated key. Same with the other method. I will try a different wording.

Reply:
Examples for each of the generic terms would go a long ways in making it more understandable, e.g.
```
Remove a value (e.g. QuantumNode or Task) from being considered
a producer of the corresponding key (e.g. DatasetRef or DatasetType).
```